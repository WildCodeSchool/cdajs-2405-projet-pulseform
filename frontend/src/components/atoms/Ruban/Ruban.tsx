import { RibbonIcon } from "../../../assets/icons/icon-list/iconList";
import type { RubanProps } from "./Ruban.type";
import "./Ruban.scss";

const Ruban = ({ days }: RubanProps) => {
  return (
    <div className="ruban">
      <div className="ruban__triangle ruban__triangle--left" />
      <div className="ruban__content">
        <RibbonIcon className="ruban__icon" />
        <span>
          {days} jour{days > 1 ? "s" : ""} d'affilée ! Continue sur ta lancée !
        </span>
      </div>
      <div className="ruban__triangle ruban__triangle--right" />
    </div>
  );
};

export default Ruban;
