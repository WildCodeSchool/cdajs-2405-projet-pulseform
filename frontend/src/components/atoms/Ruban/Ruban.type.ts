export interface RubanProps {
  days: number;
}
