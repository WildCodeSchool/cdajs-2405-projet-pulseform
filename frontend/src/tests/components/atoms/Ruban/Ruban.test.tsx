import Ruban from "@components/atoms/Ruban/Ruban";
import type { RubanProps } from "@components/atoms/Ruban/Ruban.type";
import { render } from "@testing-library/react";

describe("Ruban", () => {
  it("renders the Ruban component with a single day", () => {
    const props: RubanProps = { days: 1 };
    const tree = render(<Ruban {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("renders the Ruban component with multiple days", () => {
    const props: RubanProps = { days: 5 };
    const tree = render(<Ruban {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
