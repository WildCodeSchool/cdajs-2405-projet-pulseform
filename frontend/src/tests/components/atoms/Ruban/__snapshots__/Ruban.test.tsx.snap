// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Ruban > renders the Ruban component with a single day 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ruban"
      >
        <div
          class="ruban__triangle ruban__triangle--left"
        />
        <div
          class="ruban__content"
        >
          <span />
          <span>
            1
             jour
             d'affilée ! Continue sur ta lancée !
          </span>
        </div>
        <div
          class="ruban__triangle ruban__triangle--right"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ruban"
    >
      <div
        class="ruban__triangle ruban__triangle--left"
      />
      <div
        class="ruban__content"
      >
        <span />
        <span>
          1
           jour
           d'affilée ! Continue sur ta lancée !
        </span>
      </div>
      <div
        class="ruban__triangle ruban__triangle--right"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Ruban > renders the Ruban component with multiple days 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ruban"
      >
        <div
          class="ruban__triangle ruban__triangle--left"
        />
        <div
          class="ruban__content"
        >
          <span />
          <span>
            5
             jour
            s
             d'affilée ! Continue sur ta lancée !
          </span>
        </div>
        <div
          class="ruban__triangle ruban__triangle--right"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ruban"
    >
      <div
        class="ruban__triangle ruban__triangle--left"
      />
      <div
        class="ruban__content"
      >
        <span />
        <span>
          5
           jour
          s
           d'affilée ! Continue sur ta lancée !
        </span>
      </div>
      <div
        class="ruban__triangle ruban__triangle--right"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
